R/plot_xctr_financial.R

```diff
@@ -0,0 +1,64 @@
+#' Plot xctr data on a company or portfolio level, with financial data
+#'
+#' Plots the financial data for either a specific company or for the portfolio as a whole.
+#'
+#' @param data A data frame like [financial]
+#' @param company_name (optional) The name of the specific company to plot the
+#' financial data for. If NULL, the function will plot the financial data for the portfolio.
+#' @param mode The mode of financial data to plot.
+#' It can be one of "equal_weight", "worst_case", "best_case", or "main_activity".
+#'
+#' @return A ggplot2 object representing the financial data plot.
+#'
+#' @export
+#'
+#' @examples
+#' # Example 1: Plot financial data for a specific company
+#' plot_xctr_financial(data = financial, company_name = "peter", mode = "equal_weight")
+#'
+#' # Example 2: Plot portfolio-level financial data
+#' plot_xctr_financial(data = financial, mode = "worst_case")
+#'
+plot_xctr_financial <- function(data, company_name = NULL, mode = c("equal_weight", "worst_case", "best_case", "main_activity")) {
+  mode <- arg_match(mode)
+
+  # TODO: do we want to drop NA's everywhere silently?
+  data <- data |>
+    drop_na(-c(.data$equal_weight_finance, .data$worst_case_finance, .data$best_case_finance, .data$main_activity))
+
+  crucial <- c(
+    "main_activity",
+    "_risk_category",
+    "equal_weight_finance",
+    "worst_case_finance",
+    "best_case_finance"
+  )
+  data |> check_crucial_names(c(names(select(data, matches(crucial)))))
+
+  risk_category_var <- names(select(data, matches("_risk_category")))
+
+  data <- data |>
+    mutate(risk_category_var = factor(
+      data[[risk_category_var]],
+      levels = c("low", "medium", "high")
+    ))
```

Comment:
Here I see a concept that could be clarified with a little helper -- i.e. the idea that `risk_category_var` has values that are defined by the tilt methodology. This won't DRY the code much but might clarify that this bit of code is not casual wrangling but rather it's part of the tilt methodology. And I think such code belongs to tiltIndicator rather than here but here is a good place to start. Eventually we could move it to tiltIndicator.


``` r
as_risk_category <- function(x) {
  check_levels(x)
  factor(x, levels = risk_category_levels())
}

check_levels <- function(x) {
  stopifnot(all(sort(unique(x))  %in% sort(risk_category_levels())))
}

risk_category_levels <- function() c("low", "medium", "high")



# Example
x <- c("low", "low")
as_risk_category(x)
#> [1] low low
#> Levels: low medium high
```

<sup>Created on 2023-08-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>